Code/Framework/AzCore/AzCore/Math/IntersectSegment.h

```diff
@@ -32,32 +32,112 @@ namespace AZ
+        //! can perform less setup work.
+        //! Usage:
+        //!   SegmentTriangleHitTester hitTester(rayStart, rayEnd);
+        //!   for (auto& tri : triangles)
+        //!      if (hitTester.IntersectSegmentTriangle(tri.a, tri.b, tri.c, outNormal, outT))
+        //!         hitPoint = rayStart + (rayEnd - rayStart) * outT;
+        class SegmentTriangleHitTester
+        {
+        public:
+            //! Construct a hit tester for segment pq that can be used to compare against multiple triangles.
+            //! @param p Segment start point.
+            //! @param q Segment end point.
+            SegmentTriangleHitTester(const AZ::Vector3& p, const AZ::Vector3& q);
+
+            //! For the given segment pq and a triangle abc (CCW), returns whether the segment intersects the
+            //! triangle and if so, also returns the time of intersection along the segment and the triangle normal.
+            //! @param a Triangle vertex 1.
+            //! @param b Triangle vertex 2.
+            //! @param c Triangle vertex 3.
+            //! @param normal Triangle normal at the intersection point if segment intersects triangle.
+            //! @param t Time of intersection along the segment [0.0 (p), 1.0 (q)] if segment intersects triangle.
+            //! @return true if the segment intersects the triangle otherwise false.
+            bool IntersectSegmentTriangleCCW(
+                const AZ::Vector3& a, const AZ::Vector3& b, const AZ::Vector3& c, AZ::Vector3& normal, float& t)
+            {
+                constexpr bool oneSided = true;
+                return Intersect<oneSided>(a, b, c, normal, t);
+            }
+
+            //! Same as \ref IntersectSegmentTriangleCCW without respecting the triangle (a,b,c) vertex order (i.e. double sided).
+            //! @param a Triangle vertex 1.
+            //! @param b Triangle vertex 2.
+            //! @param c Triangle vertex 3.
+            //! @param normal Triangle normal at the intersection point if segment intersects triangle.
+            //! @param t Time of intersection along the segment [0.0 (p), 1.0 (q)] if segment intersects triangle.
+            //! @return true if the segment intersects the triangle otherwise false.
+            bool IntersectSegmentTriangle(
+                const AZ::Vector3& a, const AZ::Vector3& b, const AZ::Vector3& c, AZ::Vector3& normal, float& t)
+            {
+                constexpr bool oneSided = false;
+                return Intersect<oneSided>(a, b, c, normal, t);
+            }
+
+
+            //! Gets the segment intersection point for a given Time of intersection 't'.
+            //! @param t Time of intersection along the segment [0.0 (p), 1.0 (q)].
+            //! @return The point on the segment where the intersection occurred.
+            AZ::Vector3 GetIntersectionPoint(float t)
```

Comment:
update: missing `const` for the member function

Reply:
Fixed!